sdk/go/common/workspace/project.go

```diff
@@ -349,19 +353,41 @@ func ValidateProject(raw interface{}) error {
 		return err
 	}
 
-	// Couple of manual errors to match Validate
+	// Manually validate keys that need more validation than the raw JSON schema
+	// can provide.
+
+	// Name.
 	name, ok := project["name"]
 	if !ok {
+		closest := findClosestKey("name", project, maxValidationAttributeDistance)
+		if closest != "" {
+			return fmt.Errorf(
+				"project is missing a 'name' attribute; found '%s' instead",
+				closest,
+			)
+		}
+
 		return errors.New("project is missing a 'name' attribute")
 	}
+
 	if strName, ok := name.(string); !ok || strName == "" {
 		return errors.New("project is missing a non-empty string 'name' attribute")
 	}
+
+	// Runtime.
```

Comment:
```suggestion
```

This is pretty obvious from the code, I don't think these comments add any value here.